async-nats/src/jetstream/context.rs

```diff
@@ -90,6 +90,44 @@ impl Context {
             Response::Ok(info) => Ok(Stream { info }),
         }
     }
+
+    pub async fn delete_stream<T: AsRef<str>>(
+        &mut self,
+        stream: T,
+    ) -> Result<DeleteResponse, Error> {
```

Comment:
We'd probably want to just return a bool, ala map and sets remove?

```suggestion
    ) -> Result<bool, Error> {
```

Reply:
was thinking about that for a moment and ended up in "lets be transparent with the API". 

Rationale: nobody checks the API responses beyond errors. If they do, they might be superuser and having responses consistent with nats schemas makes things easier to work with.

Reply:
(I'm not strongly insisting on this approach btw. Just not sure which is better. This seems more explicit and self explanatory. User understands what the returned value is without checking docs)

Reply:
We can do a bit of magic for the response type maybe, implicitly into bool?

Reply:
I dont think we need that magic. Just went with bool.